server/controllers/websocket/mux.go

```diff
@@ -0,0 +1,62 @@
+package websocket
+
+import (
+	"net/http"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/logging"
+)
+
+// PartitionKeyGenerator generates partition keys for the multiplexor
+type PartitionKeyGenerator interface {
+	Generate(r *http.Request) (string, error)
+}
+
+// PartitionRegistry is the registry holding each partition
+// and is responsible for registering/deregistering new buffers
+type PartitionRegistry interface {
+	Register(key string, buffer chan string)
+	Deregister(key string, buffer chan string)
+}
+
+// Multiplexor is responsible for handling the data transfer between the storage layer
+// and the registry. Note this is still a WIP as right now the registry is assumed to handle
+// everything.
+type Multiplexor struct {
+	writer       *Writer
+	keyGenerator PartitionKeyGenerator
+	registry     PartitionRegistry
+}
+
+func NewMultiplexor(log logging.SimpleLogging, keyGenerator PartitionKeyGenerator, registry PartitionRegistry) *Multiplexor {
+	upgrader := websocket.Upgrader{}
+	upgrader.CheckOrigin = func(r *http.Request) bool { return true }
+	return &Multiplexor{
+		writer: &Writer{
+			upgrader: upgrader,
+			log:      log,
+		},
+		keyGenerator: keyGenerator,
```

Comment:
We probably need to assign the registry to the multiplexor? Don't we have a linter that checks for these kind of errors where the passed argument is not used in the method body?

Reply:
Oops, classic.  i don't think we have linters for that. Will fix.